### ISSUE TITLE
Cannot bind to Fully qualified domain name (FQDN)

### ISSUE THREAD

**Issue by croeber8** [hapijs/hapi#4477]
### Runtime

NodeJS

### Runtime version

NodeJS v18

### Module version

Hapi v21

### How can we help?

Hello, I am trying to upgrade my hapi to v21. I created a simple project just to make sure everything works. 

It seems that hapi is not binding when i use the FQDN. It only binds when using 0.0.0.0. I verified this using netstat -a. This is running NodeJS v18 and HapiJS v21 on Windows Server 2019.

How can I get HAPI to bind to the FQDN. 

Here is my code.

```
const Path = require('path')
const Fs = require('fs')
const Hapi = require('@hapi/hapi');
const host = 'DEVAZRVIRRDV03.dev.dss.local'
//0.0.0.0 works
//const host = '0.0.0.0'

async function Start() {
    try {

        const server = new Hapi.Server({
            host: host,
            port: 8080,
            // tls: {
            //     cert: Fs.readFileSync(Path.resolve('./cer.cer')),
            //     key: Fs.readFileSync(Path.resolve('./key.key'))
            // }
        });
        server.route({
            method: 'GET',
            path: '/test',
            config: {
                handler: async (request, reply) => {
                    console.log('handled')
                    return {
                        message: 'test'
                    };
                }
            }
        });
        console.log('starting')
        await server.start();
        console.log('started')
    } catch (e) {
        console.error(e)
    }
}
Start()
```

**Comment by kanongil** [comment]
Hapi doesn't do anything with the FQDN, except [pass it to the node http server `listen()`](https://github.com/hapijs/hapi/blob/23d65502a6cf5372f6db0acfaac9a5188b9050cd/lib/core.js#L339-L340) method, documented [here](https://nodejs.org/api/net.html#serverlistenport-host-backlog-callback).

FYI, the DNS resolution behaviour [has been changed in node to prefer IPv6](https://github.com/nodejs/node/pull/39987). Unlike for client connections, there doesn't seem to be any way to tell the listener it to prefer IPv4 when resolving it. You will likely have to use the `dns` module to pre-resolve it yourself.

**Comment by croeber8** [comment]
Yeah, resolving the host using dns.resolve works. Still weird I have to do that though.

![image](https://github.com/hapijs/hapi/assets/157724881/e1382587-393a-4882-88dd-6ec5ae5467b9)
